src/java/org/apache/cassandra/service/accord/AccordCommand.java

```diff
@@ -524,22 +535,29 @@ public Timestamp executeAt()
     }
 
     @Override
-    public void executeAt(Timestamp timestamp)
+    public Txn.Kind kind()
+    {
+        PartialTxn txn = partialTxn.get();
+        return txn == null ? null : txn.kind();
```

Comment:
Sure. It might also be reasonable to insist it is only invoked on commands where we know the transaction already.

Reply:
Hmm, so thinking about this more carefully, this is another unexpected side-effect of permitting `Accept` without a preceding PreAccept, as we need to know the `kind` to correctly adopt dependencies. This is probably fine, as the transaction will be visible to a quorum from its PreAccept, but a node that has not PreAccepted but has Accepted may fail to report the transaction as a dependency as a result.

I need to give it a little thought, but I am inclined to pack the `Kind` into the `TxnId` as it is simple, cheap (a single bit), and potentially simplifies other dependency analysis in future (since this would also be transmitted in all deps)

Reply:
So we shouldn't actually need this null check, since we don't register ourselves with a CommandStore until we have been pre-accepted and therefore know the transaction definition. So I think it is fine to leave this alone for now, and simply remove the null check.